File: backend/app/models/FileRow.py
# app/models/FileRow.py
from sqlalchemy import Column, Integer, ForeignKey, JSON, DateTime
from app.models import Base
from sqlalchemy.sql import func
from sqlalchemy.orm import relationship

class FileRow(Base):
    __tablename__ = 'file_rows'

    row_id = Column(Integer, primary_key=True, autoincrement=True)
    file_id = Column(Integer, ForeignKey('files.file_id'), nullable=False)
    data = Column(JSON, nullable=False)
    tag_id = Column(Integer, ForeignKey('tags.tag_id'), nullable=False)
    tag_value_id = Column(Integer, ForeignKey('tag_values.tag_value_id'), nullable=False)
    created_at = Column(DateTime, default=func.now(), nullable=False)

    file = relationship("File", back_populates="rows")
    tag = relationship("Tag", back_populates="file_rows")
    tag_value = relationship("TagValue", back_populates="file_rows")

File: backend/TagFlowApi/Models/FileRow.cs
// TagFlowApi/Models/FileRow.cs
namespace TagFlowApi.Models
{
    public class FileRow
    {
        public int RowId { get; set; }
        public string Data { get; set; } = "";
        public DateTime CreatedAt { get; set; }
        public int FileId { get; set; }
        public int TagId { get; set; }


        // Foreign key to File table
        public File File { get; set; } = null!;

        // Foreign key to Tag table
        public Tag Tag { get; set; } = null!;

        // Foreign key to TagValue table
        public int TagValueId { get; set; }
        public TagValue TagValue { get; set; } = null!;
    }
}
